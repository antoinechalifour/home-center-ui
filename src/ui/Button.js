import styled from 'styled-components'

const Button = styled.button`
  font-size: inherit;
  font-family: inherit;
  background: none;
  border: none;
  cursor: pointer;
  text-transform: uppercase;
  padding: 4px 8px;
  border-radius: 4px;
  transition: letter-spacing .25s ease-in;

  :hover {
    letter-spacing: 1px;
  }

  ${({ theme, primary }) => primary && `
    background: ${theme.colors.primary};
    color: #fff;
  `}
`

export default Button
