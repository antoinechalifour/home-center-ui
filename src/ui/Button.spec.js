import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import Button from './Button'

const theme = {
  colors: { primary: 'blueviolet' }
}

describe('ui/Button', () => {
  it('Should render a button', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Button>Add</Button>
      </ThemeProvider>
    )

    expect(tree).toMatchSnapshot()
  })

  it('Should render a primary button', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Button primary>Add</Button>
      </ThemeProvider>
    )

    expect(tree).toMatchSnapshot()
  })
})
