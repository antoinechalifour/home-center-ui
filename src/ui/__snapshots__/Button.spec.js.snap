// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Button Should render a button 1`] = `
.c0 {
  font-size: inherit;
  font-family: inherit;
  background: none;
  border: none;
  text-transform: uppercase;
  padding: 4px 8px;
  border-radius: 4px;
  font-size: 90%;
}

<button
  className="c0"
>
  Add
</button>
`;

exports[`ui/Button Should render a primary button 1`] = `
.c0 {
  font-size: inherit;
  font-family: inherit;
  background: none;
  border: none;
  text-transform: uppercase;
  padding: 4px 8px;
  border-radius: 4px;
  font-size: 90%;
  background: blueviolet;
  color: #fff;
}

<button
  className="c0"
>
  Add
</button>
`;
